src/librustdoc/html/render/mod.rs

```diff
@@ -1074,9 +1084,16 @@ fn render_assoc_items(
         let (blanket_impl, concrete): (Vec<&&Impl>, _) =
             concrete.into_iter().partition(|t| t.inner_impl().blanket_impl.is_some());
 
+        let (blanket_std, blanket_impl) =
```

Comment:
Instead of going through traits once again, wouldn't it be simpler to instead make a loop which will put into the right `vec` directly? That will allow us to not iterate over all implementations multiple times.